server/storage/backend/verify.go

```diff
@@ -67,3 +69,41 @@ func insideUnittest() bool {
 	stackTraceStr := string(debug.Stack())
 	return strings.Contains(stackTraceStr, "_test.go") && !strings.Contains(stackTraceStr, "tests/")
 }
+
+// VerifyBackendConsistency verifies data in ReadTx and BatchTx are consistent.
+func VerifyBackendConsistency(b Backend, lg *zap.Logger, skipSafeRangeBucket bool, bucket ...Bucket) {
+	verify.Verify(func() {
+		if b == nil {
+			return
+		}
+		if lg != nil {
+			lg.Debug("verifyBackendConsistency", zap.Bool("skipSafeRangeBucket", skipSafeRangeBucket))
+		}
```

Comment:
Minor comment: suggest to remove the ugly nil check. We require the callers to pass in valid parameters.
```suggestion
		lg.Debug("verifyBackendConsistency", zap.Bool("skipSafeRangeBucket", skipSafeRangeBucket))
```

Reply:
> I don't want to fail those tests.

Which tests are "those tests"?

Reply:
`TestProcessDuplicatedAppRespMessage`

Reply:
OK. Actually we shouldn't enable & run the verification in unit test at all. They are only supposed to be executed in e2e and integration tests.

We may want to remove the following setting.
https://github.com/etcd-io/etcd/blob/5d45a88ab73cc5461ccf23e4df70598d8d433ff0/scripts/test.sh#L48

Since it's unrelated to this PR, so I won't insist on it. It can be discussed separately.